pygmt/src/nearneighbor.py

```diff
@@ -0,0 +1,160 @@
+
+    {I}
+
+    region : str or list
+        *xmin/xmax/ymin/ymax*\[**+r**][**+u**\ *unit*].
+        Specify the region of interest.
+
+    search_radius : str
+        Sets the search radius that determines which data points are considered
+        close to a node.
+
+    outfile : str
+        Optional. The file name for the output netcdf file with extension .nc
+        to store the grid in.
+
+    empty : str
+        Optional. Set the value assigned to empty nodes. Defaults to NaN.
+
+    sectors : str
+        *sectors*\ [**+m**\ *min_sectors*]\|\ **n**.
+        Optional. The circular search area centered on each node is divided
+        into *sectors* sectors. Average values will only be computed if there
+        is *at least* one value inside each of at least *min_sectors* of the
+        sectors for a given node. Nodes that fail this test are assigned the
+        value NaN (but see ``empty``). If +m is omitted then *min_sectors* is
+        set to be at least 50% of *sectors* (i.e., rounded up to next integer)
+        [Default is a quadrant search with 100% coverage, i.e., *sectors* =
+        *min_sectors* = 4]. Note that only the nearest value per sector enters
+        into the averaging; the more distant points are ignored. Alternatively,
+        use ``sectors="n"`` to call GDALʻs nearest neighbor algorithm instead.
+
+    {V}
+    {a}
+    {b}
+    {d}
+    {e}
+    {f}
+    {g}
+    {h}
+    {i}
+    {r}
+    {w}
+
+    Returns
+    -------
+    ret: xarray.DataArray or None
+        Return type depends on whether the ``outfile`` parameter is set:
+
+        - :class:`xarray.DataArray`: if ``outfile`` is not set
+        - None if ``outfile`` is set (grid output will be stored in file set by
+          ``outfile``)
+    """
+
+    kind = data_kind(data, x, y, z)
+    if kind == "vectors" and z is None:
+        raise GMTInvalidInput("Must provide z with x and y.")
+
+    with GMTTempFile(suffix=".nc") as tmpfile:
+        with Session() as lib:
+            # Choose how data will be passed into the module
+            table_context = lib.virtualfile_from_data(
+                check_kind="vector", data=data, x=x, y=y, z=z
+            )
+            with table_context as infile:
+                if "G" not in kwargs.keys():  # if outfile is unset, output to tmpfile
+                    kwargs.update({"G": tmpfile.name})
+                outfile = kwargs["G"]
+                arg_str = " ".join([infile, build_arg_string(kwargs)])
+                lib.call_module(module="nearneighbor", args=arg_str)
+
+        if outfile == tmpfile.name:  # if user did not set outfile, return DataArray
+            with xr.open_dataarray(outfile) as dataarray:
+                result = dataarray.load()
+                _ = result.gmt  # load GMTDataArray accessor information
+        elif outfile != tmpfile.name:  # if user sets an outfile, return None
+            result = None
+
+    return result
```

Comment:
We've recently added a standard `load_dataarray` function at #1439 to simplify this code chunk :)

```suggestion
    return load_dataarray(outgrid) if outgrid == tmpfile.name else None
```